ibis/backends/tests/test_generic.py

```diff
@@ -166,9 +166,9 @@ def test_notin(backend, alltypes, sorted_df, column, elements):
         (lambda t: ~t['bool_col'], lambda df: ~df['bool_col']),
     ],
 )
-@pytest.mark.skip_backends(['dask'])  # TODO - sorting - #2553
+@pytest.mark.skip_backends(['dask', 'datafusion'])  # TODO - sorting - #2553
```

Comment:
```python
# bool_col is an integer here, convert it to be actually boolean
bool_col = (df.column("bool_col") == df.literal(1)).alias("bool_col")

t.select(bool_col).filter(df.column("bool_col")).collect()
```

raises

```
Exception: Error during planning: Ambiguous reference to field named 'bool_col'
```

Reply:
@kszucs i think this is a bug in datafusion, `.filter(df.column("bool_col"))` should ignore the projection alias and only use the actual `bool_col` table column to evaluate the predicate. This is be behavior I see in postgres. For example `SELECT id+1 as id2 FROM test where id2 > 2` results in `error: column "id2" does not exist`.

Do you mind filing a datafusion issue so we can dig into it?

Reply:
Thanks @houqp! Going to create an issue.

Reply:
What happened with this?